src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/update/actions/CompleteFlowPathRemovalAction.java

```diff
@@ -40,11 +40,12 @@ public CompleteFlowPathRemovalAction(PersistenceManager persistenceManager) {
     @Override
     protected void perform(State from, State to, Event event, FlowUpdateContext context, FlowUpdateFsm stateMachine) {
         Flow originalFlow = RequestedFlowMapper.INSTANCE.toFlow(stateMachine.getOriginalFlow());
+        originalFlow.setAffinityGroupId(stateMachine.getOriginalAffinityFlowGroup());
+        originalFlow.setDiverseGroupId(stateMachine.getOriginalDiverseFlowGroup());
+
         removeOldPrimaryFlowPaths(originalFlow, stateMachine);
         removeOldProtectedFlowPaths(originalFlow, stateMachine);
-
-        Flow flow = getFlow(stateMachine.getFlowId());
-        removeRejectedFlowPaths(flow, stateMachine);
+        removeRejectedFlowPaths(originalFlow, stateMachine);
```

Comment:
I this there is a bug here. Rejected Paths are new paths. And original flow doesn't contains them. That is why we had to get flow from DB with new paths. We can discuss it on some call

Reply:
Ok